tests/common.py

```diff
@@ -1815,3 +1815,20 @@ async def snapshot_platform(
         state = hass.states.get(entity_entry.entity_id)
         assert state, f"State not found for {entity_entry.entity_id}"
         assert state == snapshot(name=f"{entity_entry.entity_id}-state")
+
+
+def reset_translation_cache(hass: HomeAssistant, components: list[str]) -> None:
```

Comment:
Although I don't like this solution since it's very obscure and hard to know to use, I think it's ok for now so that we can remove the hardcoded exception for cloud below.

I like the suggestion of making `mock_integration` a context manager.